src/webcore/value.rs

```diff
@@ -51,6 +51,15 @@ impl Reference {
     }
 }
 
+impl PartialEq for Reference {
+    fn eq( &self, other: &Reference ) -> bool {
+        js!( return @{self} === @{other}; ).try_into().unwrap()
```

Comment:
Hmm.. this is probably unnecessary considering the performance hit it's going to incur.

How about something like this?

```rust
let result = self.0 == other.0;
debug_assert_eq!( js! ( return @{self} === @{other}; ).try_into().unwrap(), true );
result
```

In release mode it's going to be super cheap, while in debug mode we'll get an extra check to make sure this behaves as we think it should.

Reply:
Also, `#[inline]`.

Reply:
Good catch about `#[inline]`. But I'm confused: `js!` is used *constantly*, why are you worried about the performance *only for `PartialEq`*?

Also, I'm not sure if it actually *is* unnecessary. In issue #92 they pointed out that some JavaScript objects cannot be put into `WeakMap`, so we'll have to rethink the `Reference` implementation. And in any case relying upon `WeakMap` to guarantee correctness feels wrong to me.

Using `===` guarantees that it will ***always*** match the JavaScript behavior, without relying upon things like `WeakMap` (which might change or be refactored in the future!)

Reply:
If we have an object which cannot be put in a `WeakMap` then even when comparing it through the `js!` macro it's going to be subtly broken anyway (e.g. some of the conversion functions between Rust and JS won't work)

In general since they're so common I'd like to make `Reference`s cheaper to use in the future (e.g. by moving the refcounting into Rust https://github.com/koute/stdweb/issues/78). As for those special snowflake types which cannot be put in a `WeakMap` - I'd rather come up with some separate mechanism for those, so that only those types will incur the extra overhead instead of everything.